### PR TITLE
Create type script definitions

### PR DESCRIPTION
**Pull request by alexandrusavin**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+       */
+      requestedWith(body: any): Assertion;
+      /**
+       * true if the spy was called exactly once.
+       */
+      requestedWithHeaders(headers: object): Assertion;
+      /**
+       * true if the spy was called exactly twice.
+       */
+      requestedWithHeadersMatch(headers: object): Assertion;
```
Comment (kimroen) [inline index.d.ts]: This seems to be copy/pasted from the `sinon-chai` type definitions.

The code documentation is not correct, and the return type for these should be `Promise<Assertion>`.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as nock from 'nock';
```
Comment (kimroen) [inline index.d.ts]: This import isn't used. Is it necessary?

File: index.d.ts (hunk 1 above)
Comment (alexandrusavin) [reply index.d.ts]: It is :). I'll clean it

**Comment by chrisandrews7** [comment]
Hey @alexandrusavin, Is this nearly ready to be merged?

**Comment by aprilsourz** [comment]
@chrisandrews7 @alexandrusavin 

I'd like to see this merged as well! I'm happy to finish it up, is it best if I create my own fork?

**Comment by alexandrusavin** [comment]
Just removed the redundant import and the comments. If ok with you @chrisandrews7, then I would propose to merge this PR.

**Comment by aprilsourz** [comment]
+1 for merging this PR! For now I'm going to copy the type definitions into my project. (:

**Review by chrisandrews7 — APPROVED** [review]

**Comment by chrisandrews7** [comment]
Thanks all. Released in 1.3.0.

**Pull request merged**